source/slang/slang-emit-cpp.cpp

```diff
@@ -1561,7 +1571,117 @@ void CPPSourceEmitter::emitParamTypeImpl(IRType* type, String const& name)
 
 void CPPSourceEmitter::emitWitnessTable(IRWitnessTable* witnessTable)
 {
-    SLANG_UNUSED(witnessTable);
+    auto interfaceType = cast<IRInterfaceType>(witnessTable->getOperand(0));
+    auto witnessTableItems = witnessTable->getChildren();
+    List<IRWitnessTableEntry*> sortedWitnessTableEntries = getSortedWitnessTableEntries(witnessTable);
+    _maybeEmitWitnessTableTypeDefinition(interfaceType, sortedWitnessTableEntries);
+
+    // Define a global variable for the witness table.
+    m_writer->emit("extern ");
+    emitSimpleType(interfaceType);
+    m_writer->emit(" ");
+    m_writer->emit(getName(witnessTable));
+    m_writer->emit(";\n");
+
+    // The actual definition of this witness table global variable
+    // is deferred until the entire `Context` class is emitted, so
+    // that the member functions are available for reference.
+    // The witness table definition emission logic is defined in the
+    // `_emitWitnessTableDefinitions` function.
```

Comment:
I was initially going to say that needing to work around the `Context` type here was unfortunate because it stops the witness tables from being `static const` data, but then I realized that in the pursuit of separate compilation we would eventually need the witness tables to have `extern` linkage so that another compiled module can link against them by their mangled name.